.github/workflows/build_and_test_dbt.yaml

```diff
@@ -29,7 +29,6 @@ jobs:
         with:
           path: ${{ env.PROJECT_DIR }}/${{ env.STATE_DIR }}
           key: ${{ env.CACHE_KEY }}
-          restore-key: ${{ env.CACHE_RESTORE_KEY }}
```

Comment:
**question:** I'm slightly unclear: if we're not passing the `restore-key` to the composite action and it's not a required input, when will the fallback behavior ever be triggered?

Reply:
Good question @dfsnow! We don't expect the fallback behavior to ever be triggered as the workflow is currently written. I decided to leave it in anticipating that we may want to implement other workflows that use our custom `restore_cache` action with fallbacks in the future, but seeing as we could always remove it and reintroduce it later if we decide we need it, I'm also down to strip it out in this PR; let me know what you'd prefer!

Reply:
@jeancochrane I'm fine with leaving it in! Will approve this PR in a sec.